gradio/utils.py

```diff
@@ -1040,3 +1040,49 @@ def __setitem__(self, key: K, value: V) -> None:
 
 def get_cache_folder() -> Path:
     return Path(os.environ.get("GRADIO_EXAMPLES_CACHE", "gradio_cached_examples"))
+
+
+def diff(old, new):
```

Comment:
We should write some tests for this, and to make sure that `diff` and `apply_diff` undo each other.

Reply:
Agreed!